jdaviz/configs/default/plugins/data_menu/data_menu.py

```diff
@@ -627,6 +633,92 @@ def vue_modify_subset(self, info, *args):
         self.modify_subset(info.get('combination_mode'),
                            info.get('subset_type'))  # pragma: no cover
 
+    @observe('layer_selected')
+    def _allow_resize_subset(self, event={}):
+        # Needed during app initialization
+        if not hasattr(self, 'layer'):
+            return
+        if not len(self.layer.selected):
+            return
+
+        subset_grp = [sg for sg in self.app.data_collection.subset_groups
+                      if sg.label == self.layer.selected[0]]
+        if len(subset_grp):
+            subset = subset_grp[0]
+            if isinstance(subset.subset_state, CompositeSubsetState):
+                self.subset_resize_enabled = False
+            else:
+                self.subset_resize_enabled = True
+
+    def resize_subset(self):
+        """
+        Enable resizing of an existing subset in the viewer.
+        Detects the subset's ROI type and activates the corresponding
+        selection tool pre-populated with the existing bounds for
+        interactive resizing.
+        """
+        # future improvement: allow overriding layer.selected, with pre-validation
+        if len(self.layer.selected) != 1:
+            raise ValueError('Only one layer can be selected to resize subset.')
+        if self.layer.selected[0] not in self.existing_subset_labels:
+            raise ValueError('Selected layer is not a subset.')
```

Comment:
does this also need a check for `subset_resize_enabled` in case triggered from the API for a composite subset?